internal/backend/sessions/sessionworkflows/workflows.go

```diff
@@ -225,20 +225,14 @@ func (ws *workflows) stopped(sessionID sdktypes.SessionID) {
 		}
 	}
 
-	if err := ws.svcs.DB.UpdateSessionState(ctx, sessionID, sdktypes.NewSessionStateStopped(reason)); err != nil {
-		z.Error("update session", zap.Error(err))
-	}
+	_ = ws.updateSessionState(ctx, sessionID, sdktypes.NewSessionStateStopped(reason))
 }
 
-func (ws *workflows) errored(sessionID sdktypes.SessionID, err error, prints []string) {
-	z := ws.z.With(zap.String("session_id", sessionID.String()))
-
-	ctx, cancel := withLimitedTimeout(context.Background())
+func (ws *workflows) errored(ctx context.Context, sessionID sdktypes.SessionID, err error, prints []string) {
```

Comment:
Same 2 questions as in `stopped()` above

Reply:
answered above